custom_components/myskoda/coordinator.py

```diff
@@ -1,29 +1,51 @@
+from collections.abc import Coroutine
 from datetime import timedelta
 import logging
+from typing import Callable
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.debounce import Debouncer
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
-from homeassistant.util.ssl import get_default_context
 from myskoda import MySkoda, Vehicle
-from myskoda.event import Event, EventAccess, EventAirConditioning, ServiceEventTopic
+from myskoda.event import (
+    Event,
+    EventAccess,
+    EventAirConditioning,
+    EventOperation,
+    ServiceEventTopic,
+)
+from myskoda.models.operation_request import OperationName, OperationStatus
 from myskoda.models.user import User
 from myskoda.mqtt import EventCharging, EventType
 
-from .const import DOMAIN
+from .const import DOMAIN, FETCH_INTERVAL_IN_MINUTES, API_COOLDOWN_IN_SECONDS
 
 _LOGGER = logging.getLogger(__name__)
 
+type RefreshFunction = Callable[[], Coroutine[None, None, None]]
+
+
+class MySkodaDebouncer(Debouncer):
+    """Class to rate limit calls to MySkoda REST APIs."""
+
+    def __init__(self, hass: HomeAssistant, func: RefreshFunction) -> None:
+        """Initialize debounce."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            cooldown=API_COOLDOWN_IN_SECONDS,
+            immediate=False,
+            function=func,
+        )
+
 
 class State:
```

Comment:
Make this a dataclass, saves you a constrcutor

Reply:
Implemented in 3cac6d4